pkg/cli/log_flags.go

```diff
@@ -347,20 +348,48 @@ type settableString struct {
 	isSet bool
 }
 
-var _ flag.Value = (*settableString)(nil)
+type stringSetter struct {
+	*settableString
+}
+
+var _ flag.Value = (*stringSetter)(nil)
+var _ flag.Value = (*fileSetter)(nil)
 
 // Set implements the pflag.Value interface.
-func (l *settableString) Set(s string) error {
+func (l *stringSetter) Set(s string) error {
 	l.s = s
 	l.isSet = true
 	return nil
 }
 
 // Type implements the pflag.Value interface.
-func (l settableString) Type() string { return "<string>" }
+func (l stringSetter) Type() string { return "<string>" }
+
+// String implements the pflag.Value interface.
+func (l stringSetter) String() string { return l.s }
+
+type fileSetter struct {
+	*settableString
+	fileName string
+}
+
+// Set implements the pflag.Value interface.
+func (l *fileSetter) Set(s string) error {
```

Comment:
Confusing name.  A fileSetter doesn't set a file; it sets a string from a file.  Since they're implementing the Value interface, how about stringValue and fileContentsValue?